plugins/osdn/vnids.go

```diff
@@ -18,14 +20,104 @@ const (
 	AdminVNID = uint(0)
 )
 
+func (oc *OsdnController) GetVNID(name string, retry bool) (uint, error) {
+	oc.vnidLock.Lock()
+	defer oc.vnidLock.Unlock()
+
+	if id, ok := oc.vnidMap[name]; ok {
+		return id, nil
+	}
+
+	// Nodes asynchronously watches for both NetNamespace and service
+	// NetNamespaces populates vnid map and services/pod-setup depend on vnid map
+	// If for some reason, vnid map propagation from master to node is slow
+	// and if service/pod-setup tries to lookup vnid map then it may fail.
+	// So, this retry option will alleviate this problem.
+	if retry {
+		// Try few times up to 2 seconds
+		retries := 20
+		retryInterval := 100 * time.Millisecond
+		for i := 0; i < retries; i++ {
+			oc.vnidLock.Unlock()
```

Comment:
I'm not sure this is legit; you have a pending `defer oc.vnidLock.Unlock()`... if the time.Sleep() call panicked then you'd end up unlocking the mutex when it was already unlocked. (Which, admittedly would just cause another panic, so it's not like it's the end of the world... but if more code got added to the unlocked section later then this could be a problem.)